File: algorithms/javaalgo/src/test/java/programmers/level2/longjump/LongJumpTest.java
package programmers.level2.longjump;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertAll;

import org.junit.jupiter.api.Test;

public class LongJumpTest {

    static long[] dp;

    @Test
    void test() {
        int n1 = 4;
        int n2 = 3;

        assertAll(
            () -> assertThat(solution(n1)).isEqualTo(5),
            () -> assertThat(solution(n2)).isEqualTo(3)
        );
    }

    public long solution(int n) {
        dp = new long[2001];
        dp[1] = 1;
        dp[2] = 2;

        for (int i = 3; i < 2001; i++) {
            dp[i] = (dp[i - 2] + dp[i - 1]) % 1234567;
        }
        return dp[n];
    }
}

File: algorithms/javascriptalgo/programmers/level2/longJump.test.js
test("test", () => {
    let n1 = 4;
    let n2 = 3;

    expect(solution(n1)).toBe(5);
    expect(solution(n2)).toBe(3);
});

function solution(n) {
    let dp = [];
    dp[1] = 1;
    dp[2] = 2;
    for (let i = 3; i <= n; i++) {
        dp[i] = (dp[i - 2] + dp[i - 1]) % 1234567;
    }

    return dp[n];
}
